lib/veri_traci.mli

```diff
@@ -0,0 +1,30 @@
+open Bap.Std
+open Bap_traces.Std
+open Bil.Result
+
+type event = Trace.event
+
+class context : arch -> object('s)
```

Comment:
Looks like, that you need to put the trace itself in the context. And during the evaluation of the trace you will cut event by event from it. You can also store finished events, i.e., zipping the trace. You may even find it useful to store passed events in a reverse order (as a stack). So that at any point of time, your position in the trace is:
`e1,e2,e3,....,eN, current, eN+2, ...`